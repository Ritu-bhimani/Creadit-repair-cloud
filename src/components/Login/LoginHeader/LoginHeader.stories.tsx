import { ComponentMeta, ComponentStory } from '@storybook/react';
import { LoginHeader } from './LoginHeader';

export default {
  title: 'Login Header',
  component: LoginHeader
} as ComponentMeta<typeof LoginHeader>;

const Template: ComponentStory<typeof LoginHeader> = (args) => (
  <LoginHeader {...args} />
);

export const Primary = Template.bind({});

Primary.args = {
  altText: '',
  imageSrc: '',
  loginType: ''
};
