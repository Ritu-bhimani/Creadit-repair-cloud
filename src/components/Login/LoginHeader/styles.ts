import { css } from '@emotion/react';

export const LoginHeaderStyles = {
  root: css`
    text-align: center;
  `,
  brandImg: css`
    border: none;
    cursor: pointer;
  `,
  loginType: css`
    color: #6b7a8d;
    display: block;
    margin: 22px 0 10px;
    font-size: 16px;
  `
};
