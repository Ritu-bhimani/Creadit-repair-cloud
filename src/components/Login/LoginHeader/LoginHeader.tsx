import { Container, Grid, Typography } from '@mui/material';
import { FC } from 'react';
import { LoginHeaderStyles } from './styles';

type LoginHeaderProps = {
  imageSrc: string;
  loginType: string;
  altText?: string;
  imgLink?: string;
};

export const LoginHeader: FC<LoginHeaderProps> = (props: LoginHeaderProps) => {
  const { imageSrc, imgLink, loginType, altText } = props;
  return (
    <Container css={LoginHeaderStyles.root}>
      <Grid container direction="column">
        <a css={LoginHeaderStyles.brandImg} href={imgLink}>
          <img src={imageSrc} alt={altText} />
        </a>
        <Typography data-testid="title" css={LoginHeaderStyles.loginType}>
          {loginType}
        </Typography>
      </Grid>
    </Container>
  );
};
