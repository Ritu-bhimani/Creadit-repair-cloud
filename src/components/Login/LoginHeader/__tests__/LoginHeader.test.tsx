import { render, screen } from '@testing-library/react';
import { LoginHeader } from '../LoginHeader';

it('Login Header should match the sanpshot', () => {
  const component = render(<LoginHeader imageSrc={''} loginType={''} />);
  expect(component).toMatchSnapshot();
});

it('expect the login type rendered as Team member', () => {
  render(<LoginHeader imageSrc="" loginType="Team Member" />);
  expect(screen.getByTestId('title').textContent).toBe('Team Member');
});
