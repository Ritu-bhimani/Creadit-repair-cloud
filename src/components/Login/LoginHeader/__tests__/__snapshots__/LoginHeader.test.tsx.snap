// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Header should match the sanpshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
        >
          <a
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <img
              src=""
            />
          </a>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            data-testid="title"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <a
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <img
            src=""
          />
        </a>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          data-testid="title"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
